src/core/search/ast_expr.h

```diff
@@ -102,4 +105,9 @@ inline std::ostream& operator<<(std::ostream& os, const dfly::search::AstExpr& a
   return os;
 }
 
+inline std::ostream& operator<<(std::ostream& os, const dfly::search::AstKnnNode& ast) {
+  // os << "ast{" << ast->Debug() << "}";
```

Comment:
This is unimplemented. Also, to not pollute the global namespace (and avoid ADL on unqualified lookups) you can define it as a friend within the class `AstKnnNode` and that way a) it does not pollute lookup tables and b) you can also get rid of inline :stuck_out_tongue:

Reply:
p.s. you don't have to make this change now, and if you planted to have this unimplemented comment here and I will accept

Reply:
True, moved both